kafka-rest/src/main/java/io/confluent/kafkarest/KafkaRestConfig.java

```diff
@@ -709,31 +725,111 @@ public Properties getOriginalProperties() {
     return properties;
   }
 
+  public final Map<String, Object> getSchemaRegistryConfigs() {
+    HashMap<String, Object> possibleConfigs = new HashMap<>();
+
+    // Use a default schema.registry.url, if not specified.
+    possibleConfigs.put(SCHEMA_REGISTRY_URL_CONFIG, SCHEMA_REGISTRY_URL_DEFAULT);
+
+    possibleConfigs.putAll(originals());
+    possibleConfigs.putAll(originalsWithPrefix("schema.registry.", /* strip= */ true));
+    possibleConfigs.putAll(originalsWithPrefix("client.", /* strip= */ true));
+    possibleConfigs.putAll(originalsWithPrefix("producer.", /* strip= */ true));
+    possibleConfigs.putAll(originalsWithPrefix("consumer.", /* strip= */ true));
+
+    // Disable auto-registration of schemas. Schema registration is handled by SchemaManager.
+    possibleConfigs.put(AUTO_REGISTER_SCHEMAS, false);
+    // Disable latest-version fetching of schemas.
+    possibleConfigs.put(USE_LATEST_VERSION, false);
+
+    return Maps.filterKeys(
+        possibleConfigs, AbstractKafkaSchemaSerDeConfig.baseConfigDef().names()::contains);
+  }
+
+  public final Map<String, Object> getJsonSerializerConfigs() {
+    HashMap<String, Object> serializerConfigs = new HashMap<>();
+    serializerConfigs.putAll(originalsWithPrefix("client.", /* strip= */ true));
+    serializerConfigs.putAll(originalsWithPrefix("producer.", /* strip= */ true));
+
+    Set<String> names = new KafkaJsonSerializerConfig(serializerConfigs).values().keySet();
+
+    return Maps.filterKeys(serializerConfigs, names::contains);
+  }
+
+  public final Map<String, Object> getAvroSerializerConfigs() {
+    HashMap<String, Object> serializerConfigs = new HashMap<>();
+    serializerConfigs.putAll(originalsWithPrefix("client.", /* strip= */ true));
+    serializerConfigs.putAll(originalsWithPrefix("producer.", /* strip= */ true));
+    serializerConfigs.putAll(getSchemaRegistryConfigs());
+
+    Set<String> names = AbstractKafkaSchemaSerDeConfig.baseConfigDef().names();
+
+    return Maps.filterKeys(serializerConfigs, names::contains);
+  }
+
+  public final Map<String, Object> getJsonschemaSerializerConfigs() {
+    HashMap<String, Object> serializerConfigs = new HashMap<>();
+    serializerConfigs.putAll(originalsWithPrefix("client.", /* strip= */ true));
+    serializerConfigs.putAll(originalsWithPrefix("producer.", /* strip= */ true));
+    serializerConfigs.putAll(getSchemaRegistryConfigs());
+
+    Set<String> names =
+        Sets.union(
+            AbstractKafkaSchemaSerDeConfig.baseConfigDef().names(),
+            ImmutableSet.of(
+                KafkaJsonSchemaSerializerConfig.FAIL_INVALID_SCHEMA,
+                KafkaJsonSchemaSerializerConfig.SCHEMA_SPEC_VERSION,
+                KafkaJsonSchemaSerializerConfig.ONEOF_FOR_NULLABLES,
+                KafkaJsonSchemaSerializerConfig.JSON_INDENT_OUTPUT));
```

Comment:
nit: you could define this set in `KafkaJsonSchemaSerializerConfig`

Reply:
Barking in the wrong tree. I would also love for this set to be defined in `KafkaJsonSchemaSerializerConfig`. That's something to changed in schema-registry, not here.